src/odemis/util/test/dc_fetch_test.py

```
@@ -0,0 +1,195 @@
+            kwargs = builder.call_args.kwargs
+            self.assertEqual(kwargs["bucket_override"], "other-bucket")
+            self.assertEqual(kwargs["endpoint_override"], "https://s3.eu-west-1.amazonaws.com")
+            self.assertEqual(kwargs["region_override"], "eu-west-1")
+
+    def test_build_s3_client_uses_backend_region_as_default(self) -> None:
+        """build_s3_client_from_config should use backend._region when no override is given."""
+        from odemis.util.datacollector import S3UploadBackend, S3_REGION
+
+        backend = S3UploadBackend(
+            access_key="key",
+            secret_key="secret",
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to argument: "secret_key"

(S106)